app/Validators/LoginUserValidator.ts

```diff
@@ -0,0 +1,48 @@
+import { schema, rules } from '@ioc:Adonis/Core/Validator';
+import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext';
+
+export default class LoginUserValidator {
+    constructor(protected ctx: HttpContextContract) {}
+
+    /*
+    * Define schema to validate the "shape", "type", "formatting" and "integrity" of data.
+    *
+    * For example:
+    * 1. The username must be of data type string. But then also, it should
+    *    not contain special characters or numbers.
+    *    ```
+    *     schema.string({}, [ rules.alpha() ])
+    *    ```
+    *
+    * 2. The email must be of data type string, formatted as a valid
+    *    email. But also, not used by any other user.
+    *    ```
+    *     schema.string({}, [
+    *       rules.email(),
+    *       rules.unique({ table: 'users', column: 'email' }),
+    *     ])
+    *    ```
+    */
+    public schema = schema.create({
+        email: schema.string({}, [
+            rules.exists({ table: 'users', column: 'email' })
+        ]),
+        password: schema.string()
```

Comment:
Is this shcmea only validating towrads DB or towarst he request? Should not this be required or something?

Reply:
Or is that by default maybe?

Reply:
If you look at the register validation, it validates the input from a user, and once the register() is successful, then the login validation validates towards what in the DB based on the request input.

Reply:
When you create a validator with the **node ace:make validator CreateUser** command, it creates some files for you within the class and all you have to do is define your validation rules in **schema.create()** method.